### ISSUE TITLE
[FEATURE REQUEST] Parallel execution of the scan

### ISSUE THREAD

**Issue by renatobellotti** [autonomio/talos#332]
Is there a way to perform different scan tasks in parallel?

**Comment by mikkokotila** [comment]
See #55, #100, and #131 for more info on this. I think [Horodov](https://github.com/uber/horovod) will end up being the solution we adopt for this.

**Comment by ktokolwiek** [comment]
+1 on Horovod, I am currently trying to figure out how to use ti in hyperparameter search, will report back if I have something worth sharing.

**Comment by mikkokotila** [comment]
Wonderful. This would be a great contribution!

**Comment by bravenoob** [comment]
https://stackoverflow.com/questions/57027924/talos-multi-gpu-feature

**Comment by mikkokotila** [comment]
Parallelism is currently the most common label in the open issues, so I think there will be some action in this front soon. Maybe for 0.6.4(ish) something already.

Related: #100 #131

**Comment by awilliamson** [comment]
@mikkokotila I'm willing to help test new features if we get a branch going. I have multiple GPUs available to run model-data parallelism.

**Comment by mikkokotila** [comment]
@awilliamson perfect and thanks. I will update here once I get started.

**Comment by renatobellotti** [comment]
Have you had time to work on this yet?

**Comment by mikkokotila** [comment]
@renatobellotti not yet, but I'm getting a new machine in the next week or so, which will be a great time to get started :)

**Comment by renatobellotti** [comment]
Excellent, glad to hear that! :)

I'd like to make a suggestion: Not everybody has access to GPU clusters and/or cloud infrastructure (privacy, independence of research etc.). Therefore, it'd be very valuable to go for a flexibel approach, allowing parallel execution on both CPU and GPU clusters.
It seems to be possible to run Tensorflow sessions in parallel, perhaps this can be used in combination with a task queue library or so...

**Comment by renatobellotti** [comment]
Another thought that came to me:
Probably, the most flexible solution could be to implement the model training/evaluation tasks as jobs in a scheduling system, e. g. slurm. Then the user can provide information whether CPU/GPU nodes should be used etc.

**Comment by renatobellotti** [comment]
Has there been any progress?

**Comment by FranzHahn** [comment]
> Another thought that came to me:
> Probably, the most flexible solution could be to implement the model training/evaluation tasks as jobs in a scheduling system, e. g. slurm. Then the user can provide information whether CPU/GPU nodes should be used etc.

I think connecting to something like slurm would be super useful!

**Comment by mikkokotila** [comment]
Merging with #566
